test/health_checker_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/health_checker.h"
+#include "gtest/gtest.h"
+#include <stdio.h>
+#include <sstream>
+
+namespace google {
+
+class HealthCheckerUnittest : public ::testing::Test {
+ public:
+  HealthCheckerUnittest() {
+    std::stringstream stream("HealthCheckLocation: ''");
+    config_.ParseConfiguration(stream);
+  }
+
+  ~HealthCheckerUnittest() {
+    if (prefix_.length() > 0) {
+      std::ostringstream oss;
+      oss << prefix_ << "_metadata_agent_unhealthy";
+      std::remove(oss.str().c_str());
+
+      oss.str("");
+      oss.clear();
+      oss << prefix_ << "_kubernetes_node_thread";
+      std::remove(oss.str().c_str());
+
+      oss.str("");
+      oss.clear();
+      oss << prefix_ << "_kubernetes_pod_thread";
+      std::remove(oss.str().c_str());
+    }
+  }
+ protected:
+  void SetUnhealthy(HealthChecker healthReporter, const std::string &state_name) {
```

Comment:
Should this not be `HealthChecker& healthReporter`? Also in all other functions?
Also, the [style guide](https://google.github.io/styleguide/cppguide.html#Reference_Arguments) discourages non-`const` references, so we should make this a pointer (`HealthChecker* healthReporter`) instead.